### PR TITLE
[misc] Add containers and end-to-end result to micro-benchmarks

### PR DESCRIPTION
**Pull request by yolo2themoon**
Related issue = #4058 #4031

### CODE REVIEW

File: benchmarks/microbenchmarks/_result.py [hunk 1]
```diff
@@ -18,11 +31,17 @@ class ResultType(BenchmarkItem):
     name = 'get_result'
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_result.py:31]: Maybe change to `MeasurerType`?

Comment (yolo2themoon) [reply benchmarks/microbenchmarks/_result.py:31]: Awesome! How about renaming it as `MetricType`
it is possible to extend the observed  metrics beyond the `elapsed time` (kernel and e2e)

File: benchmarks/microbenchmarks/_plan.py [hunk 2]
```diff
@@ -49,10 +49,13 @@ def _get_kwargs(self, tags):
...
         for tag in tags:
-            if ResultType.init_taichi(arch, tag):
+            if MetricType.init_taichi(arch, tag):
                 return True
         return False
 
     def _run_func(self, tags: list):
-        return self.func(self.arch, self.basic_repeat_times,
-                         **self._get_kwargs(tags))
+        for tag in tags:
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_plan.py:57]: Maybe we can generate `kwargs` in function `run`, and then define a item `func_item` to find the function in `func_lut`, then we can get the function by `self.func_lut[kwargs[func_item]]`?

File: benchmarks/microbenchmarks/_plan.py [hunk 3]
```diff
@@ -49,10 +49,13 @@ def _get_kwargs(self, tags):
 
     def _init_taichi(self, arch, tags):
         for tag in tags:
```
Comment (lin-hitonami) [inline benchmarks/microbenchmarks/_plan.py:51]: same here

**Review by lin-hitonami — APPROVED** [review]

**Pull request merged**